lib/Sema/CSDiagnostics.h

```diff
@@ -1590,6 +1590,18 @@ class InvalidTupleSplatWithSingleParameterFailure final
   bool diagnoseAsError() override;
 };
 
+class UnnecessaryCoercionFailure final : public ContextualFailure {
```

Comment:
What be great to add a comment here just like we do for the rest of the failure about what it does and when it's used.